### PR TITLE
Failed getting entity with fetch eager property

### PR DESCRIPTION
**Pull request by dbannik**
Here, I provided tests for reproduction but without the solution to the problem itself.

Diff **WithoutFetchEager\AbstractRemoveControl** and **WithFetchEager\AbstractRemoveControl**
image

### CODE REVIEW

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 1]
```diff
@@ -3187,6 +3187,10 @@ private function eagerLoadCollections(array $collections, array $mapping): void
             foreach ($found as $targetValue) {
                 $sourceEntity = $targetProperty->getValue($targetValue);
 
+                if ($sourceEntity === null) {
+                    return;
```
Comment (dbannik) [inline lib/Doctrine/ORM/UnitOfWork.php]: Exclude collection initialization if one of the collection elements has not completed full hydration

**Comment by dbannik** [comment]
test passed. please check my fix

**Review by beberlei — DISMISSED** [review]
I didn't think about this case when working on https://github.com/doctrine/orm/pull/8391

The test and fix are valid.

I cleaned it up a little.

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @dbannik and @beberlei !

**Pull request merged**